Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -690,10 +709,16 @@ def test_module(service):
     if params.get('isFetch'):
         t = datetime.utcnow() - timedelta(hours=1)
         time = t.strftime(SPLUNK_TIME_FORMAT)
-        kwargs_oneshot = {'count': 1, 'earliest_time': time}
-        searchquery_oneshot = params['fetchQuery']
+        kwargs = {'count': 1, 'earliest_time': time}
+        query = params['fetchQuery']
         try:
-            service.jobs.oneshot(searchquery_oneshot, **kwargs_oneshot)  # type: ignore
+            for item in results.ResultsReader(service.jobs.oneshot(query, **kwargs)):  # type: ignore
+                if argToBoolean(params.get('incoming_mirror', False)):
+                    if 'event_id' not in item:
+                        return_error('Cannot mirror incidents if fetch query does not use the `notable` macro.')
+                    if not params.get('timezone'):
```

Comment:
can it be that the iterator will be empty and we wont get here and wont error on bad config?

Reply:
It can. Moved the `timezone` check out of the scope of the `for` loop.
Regarding the `event_id` field, this is the only way we can check this... as users can configure their own fetch query and make sure to enhance the notable with `event_id` with other macros, so checking that the \`notable\` string is in the fetch query isn't sufficient.